### ISSUE TITLE
Step scaling policies cannot be modified

### ISSUE THREAD

**Issue by tilgovi** [terraform-providers/terraform-provider-aws#3715]
### Terraform Version
Core - v0.11.3
AWS Provider - v1.10

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_autoscaling_policy

### Terraform Configuration Files
```hcl
resource "aws_autoscaling_policy" "ecs_general_purpose_scale_out" {
  name = "memory-reservation-high"

  autoscaling_group_name    = "${module.ecs_general_purpose.group}"
  estimated_instance_warmup = 60

  adjustment_type = "PercentChangeInCapacity"
  policy_type     = "StepScaling"

  metric_aggregation_type = "Maximum"

  step_adjustment {
    scaling_adjustment          = 100
    metric_interval_lower_bound = 0
  }
}
```

### Expected Behavior
Modifying a field, such as the scaling adjustment, should modify the resource.

### Actual Behavior
Terraform fails with a validation error.

```
Error: Error applying plan:

1 error(s) occurred:

* aws_autoscaling_policy.ecs_general_purpose_scale_out: 1 error(s) occurred:

* aws_autoscaling_policy.ecs_general_purpose_scale_out: StepScaling policy types cannot use scaling_adjustment!
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Modify a field
3. `terraform apply`

### Important Factoids
There is error is nonsensical and contradicts the documentation and the behavior on create. Step scaling policies do have a scaling adjustment.

**Comment by loivis** [comment]
@tilgovi Would you reproduce the issue and attach debug.log? I took your configuration but didn't see the error.

+ starting configuration
```
resource "aws_autoscaling_policy" "ecs_general_purpose_scale_out" {
  name = "memory-reservation-high"

  autoscaling_group_name = "${aws_autoscaling_group.test.name}"
  estimated_instance_warmup = 60

  adjustment_type = "PercentChangeInCapacity"
  policy_type     = "StepScaling"

  metric_aggregation_type = "Maximum"

  step_adjustment {
    scaling_adjustment          = 100
    metric_interval_lower_bound = 0
  }
}
```
state
```
aws_autoscaling_policy.ecs_general_purpose_scale_out:
  ID = memory-reservation-high
  provider = provider.aws
  adjustment_type = PercentChangeInCapacity
  arn = arn:aws:autoscaling:eu-west-1:123456789012:scalingPolicy:7b7cc369-7cf5-476c-aca0-f653341abf3c:autoScalingGroupName/terraform-testacc-asp-7nrzm:policyName/memory-reservation-high
  autoscaling_group_name = terraform-testacc-asp-7nrzm
  cooldown = 0
  estimated_instance_warmup = 60
  metric_aggregation_type = Maximum
  min_adjustment_step = 0
  name = memory-reservation-high
  policy_type = StepScaling
  scaling_adjustment = 0
  step_adjustment.# = 1
  step_adjustment.3610145024.metric_interval_lower_bound = 0
  step_adjustment.3610145024.metric_interval_upper_bound = 
  step_adjustment.3610145024.scaling_adjustment = 100
  target_tracking_configuration.# = 0
```
+ change `scaling_adjustment` to `10`
```
resource "aws_autoscaling_policy" "ecs_general_purpose_scale_out" {
  name = "memory-reservation-high"

  autoscaling_group_name = "${aws_autoscaling_group.test.name}"
  estimated_instance_warmup = 60

  adjustment_type = "PercentChangeInCapacity"
  policy_type     = "StepScaling"

  metric_aggregation_type = "Maximum"

  step_adjustment {
    scaling_adjustment          = 10
    metric_interval_lower_bound = 0
  }
}
```
state
```
aws_autoscaling_policy.ecs_general_purpose_scale_out:
  ID = memory-reservation-high
  provider = provider.aws
  adjustment_type = PercentChangeInCapacity
  arn = arn:aws:autoscaling:eu-west-1:123456789012:scalingPolicy:7b7cc369-7cf5-476c-aca0-f653341abf3c:autoScalingGroupName/terraform-testacc-asp-7nrzm:policyName/memory-reservation-high
  autoscaling_group_name = terraform-testacc-asp-7nrzm
  cooldown = 0
  estimated_instance_warmup = 60
  metric_aggregation_type = Maximum
  min_adjustment_step = 0
  name = memory-reservation-high
  policy_type = StepScaling
  scaling_adjustment = 0
  step_adjustment.# = 1
  step_adjustment.1339028019.metric_interval_lower_bound = 0
  step_adjustment.1339028019.metric_interval_upper_bound = 
  step_adjustment.1339028019.scaling_adjustment = 10
  target_tracking_configuration.# = 0
```

**Comment by loivis** [comment]
The error is caused by `zero` value of arguments. But only `nil` is handled in current code. I've submitted a pr and hopefully it should fix issues for other arguments in the same scenario.

**Comment by bflad** [comment]
#3739 has been merged into master and will release with v1.13.0 of the AWS provider, likely in about a week.

**Comment by bflad** [comment]
Ahead of schedule, this has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
